dashboard/app/models/contact_rollups_processed.rb

```diff
@@ -250,6 +251,32 @@ def self.extract_roles(contact_data)
     uniq_roles.blank? ? {} : {roles: uniq_roles}
   end
 
+  def self.extract_state(contact_data)
+    # The priority is: school state > user state > form geo state
+    # US state in schools table is in abbreviation, must convert it back to state name.
+    school_state = extract_field_latest_value contact_data, 'dashboard.schools', 'state'
```

Comment:
A user will only have one row here...why use latest value here and not in other processing steps?

Reply:
I'm trying to not use too many implicit assumptions here. We join `users` to `school_infos` then to `schools` to map an email  (not user_id) to a school. There is a chance that an email is mapped to more than 1 school.

Reply:
This query returns 59 users with more than 1 school. It's a small number, but it means our data can sometime surprise us.
```sql
SELECT t.email, count(*) as hit, count(distinct s.id) as distinct_school
FROM schools AS s
INNER JOIN school_infos AS si ON s.id = si.school_id
INNER JOIN users AS t ON si.id = t.school_info_id
WHERE email > ''
GROUP BY email
HAVING distinct_school > 1
ORDER BY distinct_school desc
```